aten/src/ATen/native/cuda/SegmentReduce.cu

```diff
@@ -52,6 +79,11 @@ Tensor _segment_reduce_cuda_kernel(
   auto offsets = _get_complete_sum(lengths);
```

Comment:
I forgot, is it checked somewhere that lengths are contiguous? Otherwise this won't work.

Reply:
yes, caller on .cpp file is doing both for cuda and cpu:

https://github.com/pytorch/pytorch/blob/0054702491e7ba8fcdc0ff7b87760782497841cc/aten/src/ATen/native/SegmentReduce.cpp#L180